File: Net/Patterns/State/After/SoldOutState.cs
namespace Patterns.State.After
{
    using System;

    public class SoldOutState : IState
    {
        public void Insert()
        {
            Console.WriteLine(MachineMessages.InsertWhenSoldOut);
        }

        public void Eject()
        {
            Console.WriteLine(MachineMessages.EjectWhenSoldOut);
        }

        public void TurnCrank()
        {
            Console.WriteLine(MachineMessages.TurnWhenSoldOut);
        }

        public void Dispense()
        {
            Console.WriteLine(MachineMessages.DispenseWhenSoldOut);
        }
    }
}

File: Java/After/src/main/state/SoldOutState.java
package state;

public class SoldOutState implements State {
    GumballMachine gumballMachine;
 
    public SoldOutState(GumballMachine gumballMachine) {
        this.gumballMachine = gumballMachine;
    }
 
	public void insertQuarter() {
		System.out.println(MachineMessages.InsertWhenSoldOut);
	}
 
	public void ejectQuarter() {
		System.out.println(MachineMessages.EjectWhenSoldOut);
	}
 
	public void turnCrank() {
		System.out.println(MachineMessages.TurnWhenSoldOut);
	}
 
	public void dispense() {
		System.out.println(MachineMessages.DispenseWhenSoldOut);
	}
}
